cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -139,27 +152,30 @@ fn insert_cmp(
     let res = if opcode == Opcode::Fcmp {
         let cc = *fgen.u.choose(FloatCC::all())?;
 
-        // Some FloatCC's are not implemented on AArch64, see:
-        // https://github.com/bytecodealliance/wasmtime/issues/4850
         // We filter out condition codes that aren't supported by the target at
         // this point after randomly choosing one, instead of randomly choosing a
         // supported one, to avoid invalidating the corpus when these get implemented.
-        if matches!(fgen.target_triple.architecture, Architecture::Aarch64(_))
-            && ![
-                FloatCC::Ordered,
-                FloatCC::Unordered,
-                FloatCC::Equal,
-                FloatCC::NotEqual,
-                FloatCC::LessThan,
-                FloatCC::LessThanOrEqual,
-                FloatCC::GreaterThan,
-                FloatCC::GreaterThanOrEqual,
-            ]
-            .contains(&cc)
-        {
-            return Err(arbitrary::Error::IncorrectFormat.into());
+        let unimplemented_ccs = match (fgen.target_triple.architecture, args[0]) {
+            // Some FloatCC's are not implemented on AArch64, see:
+            // https://github.com/bytecodealliance/wasmtime/issues/4850
+            (Architecture::Aarch64(_), _) => &[
+                FloatCC::OrderedNotEqual,
+                FloatCC::UnorderedOrEqual,
+                FloatCC::UnorderedOrLessThan,
+                FloatCC::UnorderedOrLessThanOrEqual,
+                FloatCC::UnorderedOrGreaterThan,
+                FloatCC::UnorderedOrGreaterThanOrEqual,
+            ][..],
+            (Architecture::X86_64, ty) if ty.is_vector() => {
+                &[FloatCC::UnorderedOrEqual, FloatCC::OrderedNotEqual][..]
+            }
+            _ => &[],
         };
 
+        if unimplemented_ccs.contains(&cc) {
```

Comment:
What do you think of this style? (It may need help from `rustfmt`, I'm just typing things straight into this comment.)
```suggestion
        let unimplemented_cc = match (fgen.target_triple.architecture, cc) {
            // Some FloatCC's are not implemented on AArch64, see:
            // https://github.com/bytecodealliance/wasmtime/issues/4850
            (Architecture::Aarch64(_),
                FloatCC::OrderedNotEqual |
                FloatCC::UnorderedOrEqual |
                FloatCC::UnorderedOrLessThan |
                FloatCC::UnorderedOrLessThanOrEqual |
                FloatCC::UnorderedOrGreaterThan |
                FloatCC::UnorderedOrGreaterThanOrEqual) => true,
            (Architecture::X86_64, FloatCC::UnorderedOrEqual | FloatCC::OrderedNotEqual) if args[0].is_vector() => true,
            _ => false,
        };

        if unimplemented_cc {
```

Reply:
I don't really mind that, but the rustfmt format for that makes it quite hard to parse for me, what do you think?
```
        let unimplemented_cc = match (fgen.target_triple.architecture, cc) {
            // Some FloatCC's are not implemented on AArch64, see:
            // https://github.com/bytecodealliance/wasmtime/issues/4850
            (
                Architecture::Aarch64(_),
                FloatCC::OrderedNotEqual
                | FloatCC::UnorderedOrEqual
                | FloatCC::UnorderedOrLessThan
                | FloatCC::UnorderedOrLessThanOrEqual
                | FloatCC::UnorderedOrGreaterThan
                | FloatCC::UnorderedOrGreaterThanOrEqual,
            ) => true,
            (Architecture::X86_64, FloatCC::UnorderedOrEqual | FloatCC::OrderedNotEqual)
                if args[0].is_vector() =>
            {
                true
            }
            _ => false,
        };
        if unimplemented_cc {
```

Reply:
Hmm, yeah, that's... not ideal? It's just that I'd prefer to use `match` instead of `slice::contains` to let the compiler decide how to implement this check. I guess for me that preference slightly outweighs style preferences, but I'm willing to be persuaded otherwise.

Two things we could try which might improve the style:
- Put each arch/cc combination in a separate match arm, e.g. `(Architecture::Aarch64(_), FloatCC::OrderedNotEqual) => true` instead of using or-patterns
- Replace `if x => true` with `=> x`